v2/dht.go

```diff
@@ -31,6 +33,13 @@ type DHT struct {
 	// rt holds a reference to the routing table implementation. This can be
 	// configured via the Config struct.
 	rt kad.RoutingTable[key.Key256, kad.NodeID[key.Key256]]
+
+	// --- go-kademlia specific fields below ---
+
+	// qSubs tracks a mapping of queries to their subscribers
+	qSubs   map[query.QueryID]chan<- kad.Response[key.Key256, ma.Multiaddr]
+	qSubsLk sync.RWMutex
+	qSubCnt uint64
```

Comment:
I still think this would better handled in go-kademlia

Reply:
Yeah it would be nicer but the compromise is that the logic for stopping the query in the event loop has to be injected